workflow/scripts/heatmap.r

```diff
@@ -0,0 +1,127 @@
+###############################################################################
+#
+#R script to generate a heatmap
+#
+#   AUTHOR: Krish Agarwal
+#   AFFILIATION: University_of_Basel
+#   CONTACT: akrish136@gmail.com
+#   CREATED: 23-07-2020
+#   LICENSE: Apache_2.0
+##############################################################################
+
+# by default: suppress warnings
+options(warn = -1)
+
+# load libraries
+suppressPackageStartupMessages(library("optparse"))
+library(ggplot2)
+
+# list the command-line arguments
+option_list <- list(
+  make_option(c("--input_tsv"),
+    action = "store",
+    dest = "input_tsv",
+    type = "character",
+    help = "location and name of the tsv file"
+  ),
+  make_option(c("--input_sequence"),
+    action = "store",
+    dest = "input_sequence",
+    type = "character",
+    help = "full input_sequence"
+  ),
+  make_option(c("--outfile"),
+    action = "store",
+    dest = "output_tsv",
+    type = "character",
+    help = "location and name of output heatmap"
+  )
+)
+
+# parse command-line arguments
+opt_parser <- OptionParser(
+  usage = "Usage: %prog [OPTIONS] --message [STRING]",
+  option_list = option_list,
+  add_help_option = FALSE,
+  description = ""
+)
+opt <- parse_args(opt_parser)
+
+
+input_sequence <- opt$input_sequence
+input_tsv <- opt$input_tsv
+output_tsv <- opt$output_tsv
+
+x_axis_numbers = 1:nchar(input_sequence)
+
+file = file(input_tsv, "r")
+
+y_axis_labels = c()
+
+while (TRUE) 
+{
+  line = readLines(file, n = 1)
+  if (length(line) == 0) 
+  {
+    break
+  }
```

Comment:
I will accept this structure since there is nothing technically wrong with it but please be aware that while(true) -> if:break loops are bad algorithmic practice.

Take a look here:
https://softwareengineering.stackexchange.com/questions/142144/whiletrue-and-loop-breaking-anti-pattern
https://www.quora.com/Is-there-anything-bad-about-using-a-while-true-loop-and-exiting-it-using-a-break-statement

I am sure that `R` has a better build-in mechanism of reading files than line-by-line. Not a very pretty coding style...